skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceListener1.scala

```diff
@@ -0,0 +1,77 @@
+package com.evolutiongaming.skafka.consumer
+
+import cats.data.NonEmptySet
+import cats.~>
+import com.evolutiongaming.skafka.TopicPartition
+
+/**
+  * Will be converted to [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]] during consumer.subscribe
+  *
+  * Uses [[RebalanceCallback]] to describe the actions to be performed during rebalance.
+  *
+  * Please refer to [[RebalanceCallback]] for more details.
+  *
+  * Below is an example inspired by [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener]] documentation.
+  *
+  * Compiling and working example is available [[https://github.com/evolution-gaming/skafka/blob/d2af038b012523533f2b73d432721d6d1e7cebbe/skafka/src/test/scala/com/evolutiongaming/skafka/consumer/RebalanceListener1Spec.scala here]]:
+  * TODO: replace with shorter github link after merge of https://github.com/evolution-gaming/skafka/pull/122
+  * {{{
+  *
+  * class SaveOffsetsOnRebalance[F[_]: Applicative] extends RebalanceListener1[F] {
+  *  import RebalanceCallback._
+  *
+  *  def onPartitionsAssigned(partitions: Nes[TopicPartition]) =
+  *    for {
+  *      // read the offsets from an external store using some custom code not described here
+  *      offsets <- lift(readOffsetsFromExternalStore[F](partitions))
+  *      _       <- offsets.toList.traverse_ { case (partition, offset) => seek(partition, offset) }
```

Comment:
please **avoid** using `traverse_ `, use `foldMapA/foldMapM` instead

Reply:
please also specify variable like `a` and return it in `yield`

Reply:
Same rule should be applied everywhere else in the codebase

Reply:
fixed in 61ff053c